omero/annotation_scripts/Convert_KeyVal_namespace.py

```diff
@@ -0,0 +1,333 @@
+    "Tag": ["Project", "Dataset", "Image",
+            "Screen", "Plate", "Well", "Acquisition"]
+}
+
+P_DTYPE = "Data_Type"  # Do not change
+P_IDS = "IDs"  # Do not change
+P_TARG_DTYPE = "Target Data_Type"
+P_OLD_NS = "Old Namespace (blank for default)"
+P_NEW_NS = "New Namespace (blank for default)"
+P_MERGE = "Create new and merge"
+
+
+def get_children_recursive(source_object, target_type):
+    if CHILD_OBJECTS[source_object.OMERO_CLASS] == target_type:
+        # Stop condition, we return the source_obj children
+        if source_object.OMERO_CLASS != "WellSample":
+            return source_object.listChildren()
+        else:
+            return [source_object.getImage()]
+    else:  # Not yet the target
+        result = []
+        for child_obj in source_object.listChildren():
+            # Going down in the Hierarchy list
+            result.extend(get_children_recursive(child_obj, target_type))
+        return result
+
+
+def target_iterator(conn, source_object, target_type, is_tag):
+    if target_type == source_object.OMERO_CLASS:
+        target_obj_l = [source_object]
+    elif source_object.OMERO_CLASS == "PlateAcquisition":
+        # Check if there is more than one Run, otherwise
+        # it's equivalent to start from a plate (and faster this way)
+        plate_o = source_object.getParent()
+        wellsamp_l = get_children_recursive(plate_o, "WellSample")
+        if len(list(plate_o.listPlateAcquisitions())) > 1:
+            # Only case where we need to filter on PlateAcquisition
+            run_id = source_object.getId()
+            wellsamp_l = filter(lambda x: x._obj.plateAcquisition._id._val
+                                == run_id, wellsamp_l)
+        target_obj_l = [wellsamp.getImage() for wellsamp in wellsamp_l]
+    elif target_type == "PlateAcquisition":
+        # No direct children access from a plate
+        if source_object.OMERO_CLASS == "Screen":
+            plate_l = get_children_recursive(source_object, "Plate")
+        elif source_object.OMERO_CLASS == "Plate":
+            plate_l = [source_object]
+        target_obj_l = [r for p in plate_l for r in p.listPlateAcquisitions()]
+    elif is_tag:
+        target_obj_l = conn.getObjectsByAnnotations(target_type,
+                                                    [source_object.getId()])
+        # Need that to load objects
+        obj_ids = [o.getId() for o in target_obj_l]
+        target_obj_l = list(conn.getObjects(target_type, obj_ids))
+    else:
+        target_obj_l = get_children_recursive(source_object,
+                                              target_type)
+
+    print(f"Iterating objects from {source_object}:")
+    for target_obj in target_obj_l:
+        print(f"\t- {target_obj}")
+        yield target_obj
+
+
+def main_loop(conn, script_params):
+    """
+    For every object:
+     - Find annotations in the namespace
+     - Remove annotations with old namespace
+     - Create annotations with new namespace
```

Comment:
This logic is the behaviour if `merge` is True, but otherwise we just update the ns, right?

Reply:
Yes that's the behaviour with merge=True, otherwise it changes the namespace (default is just change namespace). 
I changed the description to reflect that